### PR TITLE
Add guest_devices support to the API

### PR DESCRIPTION
**Pull request by skovic**
This PR adds guest_devices support to the API.

### CODE REVIEW

**Comment by skovic** [comment]
@miq-bot add_label wip

**Comment by skovic** [comment]
@miq-bot remove_label wip

**Comment by skovic** [comment]
@abellotti Can we get this PR merged now that https://github.com/ManageIQ/manageiq/pull/15371 has been merged? Thanks

**Comment by abellotti** [comment]
This LGTM!!.  @imtayadeway quick 👀 when you get a chance ?  Thanks.

File: spec/requests/api/guest_devices_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        run_get(guest_devices_url(device.id))
+
+        expect_single_resource_query("device_name" => "Broadcom 2-port 1GbE NIC Card",
+                                     "device_type" => "ethernet",
+                                     "location"    => "Bay 7")
+      end
+    end
+
+    context "with an invalid role" do
```
Comment (imtayadeway) [inline spec/requests/api/guest_devices_spec.rb]: Nit, but wording here is misleading. It's not that the role is invalid, it's that the user's role does not authorize them to use this feature. Similarly, it doesn't fail - it does the right thing! i.e. respond with a forbidden status. Let's merge this as-is though, maybe you could consider rewording this in a follow up?

**Comment by skovic** [comment]
@imtayadeway I reworded the descriptions in the test file. Let me know if it looks OK now. Also, I noticed that guest_devices_spec.rb was in spec/requests/api/ instead of spec/requests/ so I moved it to spec/requests/ -- this must have happened while migrating these changes to this repo.

**Comment by imtayadeway** [comment]
@skovic thanks for making those changes! 😻 This looks great - it's just failing because of some upstream breakages - I'll restart your build once those are under control.

**Comment by juliancheal** [comment]
@imtayadeway yeah still looks like it's failing, what's the upstream issue causing it?

**Comment by rodneyhbrown7** [comment]
@juliancheal , @blomquisg

**Comment by abellotti** [comment]
Thanks @skovic for the API Enhancement !!   👍

**Pull request merged**